### PR TITLE
Use overrides from @JsonProperty when it's provided

### LINKED ISSUE #21962: fix/Spring Boot SDK: do not ignore json field names from annotations
**Description**

A fix for a bug that is also present in the SDK here has been implemented: https://github.com/camunda-community-hub/spring-zeebe/pull/934

Original issue https://github.com/camunda-community-hub/spring-zeebe/issues/903

This fix should also be done here.

### PR DESCRIPTION
**Pull request by ana-vinogradova-camunda**
## Description

Use overrides from @JsonProperty when it's provided.

This is a bug fix for the following issue https://github.com/camunda-community-hub/spring-zeebe/issues/903 that was also present in the official SDK.
A fix for the same bug in the community project : https://github.com/camunda-community-hub/spring-zeebe/pull/934

What was done:
- Added field mapping from JsonProperty annotation
- Added test to verify the bugfix

## Checklist

- [ ] for CI changes:
  - [ ] structural/foundational changes signed off by [CI DRI](https://github.com/cmur2)
  - [ ] [ci.yml](https://github.com/camunda/camunda/blob/main/.github/workflows/ci.yml) modifications comply with ["Unified CI" requirements](https://github.com/camunda/camunda/wiki/CI-&-Automation#workflow-inclusion-criteria)

## Related issues

closes #21962

### CODE REVIEW

File: clients/spring-boot-starter-camunda-sdk/src/main/java/io/camunda/zeebe/spring/client/properties/PropertyBasedZeebeWorkerValueCustomizer.java [hunk 1]
```diff
@@ -117,10 +119,17 @@ private List<String> readVariablesAsTypeParameters(final MethodInfo methodInfo)
     parameters.forEach(
         pi ->
             ReflectionUtils.doWithFields(
-                pi.getParameterInfo().getType(), f -> result.add(f.getName())));
+                pi.getParameterInfo().getType(), f -> result.add(extractParameterName(f))));
     return result;
   }
 
+  private String extractParameterName(final Field field) {
```
Comment (mustafadagher) [inline clients/spring-boot-starter-camunda-sdk/src/main/java/io/camunda/zeebe/spring/client/properties/PropertyBasedZeebeWorkerValueCustomizer.java:126]: 🔧 nitpicking: I'd have named this `extractFieldName`

**Review by mustafadagher — APPROVED** [review]
LGTM! 🚀 
only I minor suggestion below, but feel free to ignore it.

File: clients/spring-boot-starter-camunda-sdk/src/test/java/io/camunda/zeebe/spring/client/properties/PropertyBasedZeebeWorkerValueCustomizerTest.java [hunk 2]
```diff
@@ -407,4 +426,9 @@ public void setVar4(final String var4) {
       this.var4 = var4;
     }
   }
+
+  private static class PropertyAnnotatedClass {
```
Comment (mustafadagher) [inline clients/spring-boot-starter-camunda-sdk/src/test/java/io/camunda/zeebe/spring/client/properties/PropertyBasedZeebeWorkerValueCustomizerTest.java:430]: final

File: clients/spring-boot-starter-camunda-sdk/src/main/java/io/camunda/zeebe/spring/client/properties/PropertyBasedZeebeWorkerValueCustomizer.java (hunk 1 above)
Comment (ana-vinogradova-camunda) [reply clients/spring-boot-starter-camunda-sdk/src/main/java/io/camunda/zeebe/spring/client/properties/PropertyBasedZeebeWorkerValueCustomizer.java:126]: Sounds better, yes 👍

File: clients/spring-boot-starter-camunda-sdk/src/main/java/io/camunda/zeebe/spring/client/properties/PropertyBasedZeebeWorkerValueCustomizer.java [hunk 3]
```diff
@@ -117,10 +119,17 @@ private List<String> readVariablesAsTypeParameters(final MethodInfo methodInfo)
...
         pi ->
             ReflectionUtils.doWithFields(
-                pi.getParameterInfo().getType(), f -> result.add(f.getName())));
+                pi.getParameterInfo().getType(), f -> result.add(extractFieldName(f))));
     return result;
   }
 
+  private String extractFieldName(final Field field) {
+    if (field.isAnnotationPresent(JsonProperty.class)) {
+      return field.getAnnotation(JsonProperty.class).value();
```
Comment (mustafadagher) [inline clients/spring-boot-starter-camunda-sdk/src/main/java/io/camunda/zeebe/spring/client/properties/PropertyBasedZeebeWorkerValueCustomizer.java:128]: ❌ Actually, one final observation. The annotation could be present but the `value` could still be empty. resulting in a wrong field name.

We should make this check and add an extra test with `@JsonProperty` without a value

**Review by mustafadagher — CHANGES_REQUESTED** [review]

**Comment by megglos** [comment]
@ana-vinogradova-camunda should we consider backporting this to 8.5/8.6?

**Comment by ana-vinogradova-camunda** [comment]
> @ana-vinogradova-camunda should we consider backporting this to 8.5/8.6?

Yes. I will backport it.

**Review by mustafadagher — APPROVED** [review]
Great work! Thanks a lot Ana!

**Pull request merged**